DSharpPlus.SlashCommands/SlashCommandsExtension.cs

```diff
@@ -827,12 +827,13 @@ private async Task<List<object>> ResolveInteractionCommandParameters(Interaction
                 var parameter = parameters.ElementAt(i);
 
                 //Accounts for optional arguments without values given
-                if (parameter.IsOptional && (options == null ||
-                                             (!options?.Any(x => x.Name == parameter.GetCustomAttribute<OptionAttribute>().Name.ToLower()) ?? true)))
+                if (parameter.IsOptional && (!options?.Any(x =>
+                        x.Name.Equals(parameter.GetCustomAttribute<OptionAttribute>().Name, StringComparison.InvariantCultureIgnoreCase)) ?? true))
                     args.Add(parameter.DefaultValue);
                 else
                 {
-                    var option = options.Single(x => x.Name == parameter.GetCustomAttribute<OptionAttribute>().Name.ToLower());
+                    var option = options.Single(x =>
+                        x.Name.Equals(parameter.GetCustomAttribute<OptionAttribute>().Name, StringComparison.InvariantCultureIgnoreCase));
```

Comment:
This case is done tested through testing via https://github.com/DSharpPlus/DSharpPlus/blob/ba40ee3ef95bfb1d7a46b3f6f149b603593c6bb3/DSharpPlus.Test/SlashCommandMinMaxTest.cs#L33 
As seen from debugger on screenshot option is getting parsed correctly
![image](https://user-images.githubusercontent.com/50947930/152699334-8521af59-f5db-4871-a2c1-bc12cee600ed.png)